r/R/csv.R

```diff
@@ -172,8 +172,30 @@ csv_table_reader.default <- function(file,
 #'
 #' Use arrow::csv::TableReader from [csv_table_reader()]
 #'
-#' @param ... Used to construct an arrow::csv::TableReader
+#' @inheritParams csv_table_reader
+#'
+#' @param col_select [tidy selection specification][tidyselect::vars_select] of columns
+#' @param as_tibble Should the [arrow::Table][arrow__Table] be converted to a data frame.
+#'
 #' @export
-read_csv_arrow <- function(...) {
-  csv_table_reader(...)$Read()
+read_csv_arrow <- function(file,
+  read_options = csv_read_options(),
+  parse_options = csv_parse_options(),
+  convert_options = csv_convert_options(),
+  col_select = NULL,
+  as_tibble = TRUE
+  )
+{
+  reader <- csv_table_reader(file,
+    read_options = read_options,
+    parse_options = parse_options,
+    convert_options = convert_options)
+
+  tab <- reader$Read()$select(!!enquo(col_select))
```

Comment:
Since this and read_json use $Read() then $select(), it would probably make sense to always mmap (https://issues.apache.org/jira/browse/ARROW-5502). Otherwise, IIUC, the readers will read the entire file into memory in Arrow before selecting the columns, which defeats the purpose of only selecting a few columns.

Reply:
I've switched to `mmap_open()` but I don't think that makes much difference here anyway, the csv reader has no way as far as I understand to only read selected columns, so if we do want `col_select()` it's after the `Table` with all the columns is materialized anyway.

Reply:
Alright, well, even so, I would expect the csv reader to grow better memory mapping support for this as the C++ Datasets API progresses, so we'll be ready for that.